### PR TITLE
Incoming

### PR DESCRIPTION
**Pull request by sternomer**

### CODE REVIEW

**Review by MichaelSimkin — CHANGES_REQUESTED** [review]
Things to do:
1) Route for getting multiple files (by filters)
2) Comments (in manager)
3) Tests

File: src/express/incoming/manager.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import * as mongoose from 'mongoose';
+import { IIncomingFile, INewIncomingFile } from './interface';
+import IncomingFileModel from './model';
+
+/**
+ * Create a new feature.
+ * @param {INewFeature} feature - The feature to create.
+ * @returns {Promise<IFeature>} - Promise object containing the created feature.
+ */
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:9]: Delete old lines ...

File: src/express/incoming/manager.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * @param {INewIncomingFile} file - the file to create
+ * @returns {IIncomingFile} - Promise containing the incoming file
+ */
+export const createIncomingFile = (file: INewIncomingFile): Promise<IIncomingFile> => {
+    return IncomingFileModel.create(file);
+};
+
+/**
+ * Get incoming file.
+ * @param  fileId - id of current file .
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:22]: formatting

```suggestion
 * @param fileId - id of current file.
```

File: src/express/incoming/manager.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * @returns {Promise<IIncomingFile>} - Promise object containing the file with current id.
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:23]:
```suggestion
 * @returns {Promise<IIncomingFile>} Promise object containing the file with current id.
```

File: src/express/incoming/manager.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * Create a new incoming file
+ * @param {INewIncomingFile} file - the file to create
+ * @returns {IIncomingFile} - Promise containing the incoming file
+ */
+export const createIncomingFile = (file: INewIncomingFile): Promise<IIncomingFile> => {
+    return IncomingFileModel.create(file);
+};
+
+/**
+ * Get incoming file.
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:21]:
```suggestion
 * Get incoming File.
```

File: src/express/incoming/manager.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import IncomingFileModel from './model';
+
+/**
+ * Create a new feature.
+ * @param {INewFeature} feature - The feature to create.
+ * @returns {Promise<IFeature>} - Promise object containing the created feature.
+ */
+
+/**
+ * Create a new incoming file
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:12]:
```suggestion
 * Create a new incoming File.
```

File: src/express/incoming/manager.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    if (!result) throw new Error('Incoming file not found');
+    return result;
+};
+
+/**
+ * Get incoming file.
+ * @param {Partial<IIncomingFile>} query - The query to filter the files.
+ * @returns {Promise<IIncomingFile[]>} - Promise object containing the filtered files.
+ */
+
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:36]: remove unneeded line

```suggestion
```

File: src/express/incoming/manager.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+export const getMultipleFiles = async (query: any): Promise<IIncomingFile[]> => {
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:37]: Why not name the function `getIncomingFiles`?

```suggestion
export const getIncomingFiles = async (query: any): Promise<IIncomingFile[]> => {
```

File: src/express/incoming/manager.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+export const getIncomingFileById = async (fileId: mongoose.Types.ObjectId): Promise<IIncomingFile> => {
+    const result = await IncomingFileModel.findOne({ _id: fileId }).exec();
+    if (!result) throw new Error('Incoming file not found');
+    return result;
+};
+
+/**
+ * Get incoming file.
+ * @param {Partial<IIncomingFile>} query - The query to filter the files.
+ * @returns {Promise<IIncomingFile[]>} - Promise object containing the filtered files.
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:34]:
```suggestion
 * @returns {Promise<IIncomingFile[]>} Promise object containing the filtered files.
```

File: src/express/incoming/manager.ts (hunk 7 above)
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:37]: query shouldn't be `any`. Set its type to `IIncomingFileFilters` and create it in interface.ts

File: src/express/incoming/manager.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+export const getMultipleFiles = async (query: any): Promise<IIncomingFile[]> => {
+    return IncomingFileModel.find(query).exec();
+};
+
+/**
+ *
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:42]: Add description.

File: src/express/incoming/manager.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+ * @param fileId - id of current file
+ * @returns {IIncomingFile} - Promise object containing the deleted file
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:44]:
```suggestion
 * @returns {IIncomingFile} Promise object containing the deleted file.
```

File: src/express/incoming/manager.ts [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+ * @returns {Promise<IIncomingFile[]>} - Promise object containing the filtered files.
+ */
+
+export const getMultipleFiles = async (query: any): Promise<IIncomingFile[]> => {
+    return IncomingFileModel.find(query).exec();
+};
+
+/**
+ *
+ * @param fileId - id of current file
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:43]:
```suggestion
 * @param fileId - id of current file.
```

File: src/express/incoming/manager.ts [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+ * @returns {IIncomingFile} - Promise object containing the deleted file
+ */
+
```
Comment (MichaelSimkin) [inline src/express/incoming/manager.ts:46]: Delete unneeded line.
```suggestion
```

File: src/express/incoming/router.ts [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+incomingRouter.get(
+    '/:fileId',
+    ValidateRequest(incomingValidator.getIncomingFileByIdRequestSchema),
+    wrapMiddleware(incomingFileController.getIncomingFileById),
+);
+
+incomingRouter.get(
+    '/',
+    ValidateRequest(incomingValidator.getMultipleFilesRequestSchema),
+    wrapMiddleware(incomingFileController.getMultipleFiles),
```
Comment (MichaelSimkin) [inline src/express/incoming/router.ts:24]: Change names:
```suggestion
    ValidateRequest(incomingValidator.getIncomingFilesRequestSchema),
    wrapMiddleware(incomingFileController.getIncomingFiles),
```

File: tests/incoming.test.ts [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+    });
+
+    afterEach(async () => {
+        await removeIncomingCollection();
+    });
+
+    afterAll(async () => {
+        await mongoose.disconnect();
+    });
+    const incomingFile = {
```
Comment (MichaelSimkin) [inline tests/incoming.test.ts:27]: Add new line here.
```suggestion

    const incomingFile = {
```

File: tests/incoming.test.ts [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+                .post('/api/incoming')
+                .send({
+                    ...incomingFile,
+                })
+                .expect(200);
+            await request(app).get(`/api/incoming/swef`);
+            expect(400);
+        });
+    });
+    describe('get multiple files', () => {
```
Comment (MichaelSimkin) [inline tests/incoming.test.ts:74]: Add new line

```suggestion

    describe('get multiple files', () => {
```

File: tests/incoming.test.ts [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+                .get('/api/incoming')
+                .query({ name: createdFile.name })
+                .expect(200);
+            expect(resultWithQuerry.length).toBe(1);
+
+            const { body: resultWithOutQuerry } = await request(app).get('/api/incoming').expect(200);
+            expect(resultWithOutQuerry.length).toBe(2);
+        });
+    });
+    describe('delete incoming file', () => {
```
Comment (MichaelSimkin) [inline tests/incoming.test.ts:99]: Add new line.
```suggestion

    describe('delete incoming file', () => {
```

File: tests/incoming.test.ts [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+                .send({
+                    ...incomingFile,
+                })
+                .expect(200);
+            await request(app).get(`/api/incoming/swef`);
+            expect(400);
+        });
+    });
+    describe('get multiple files', () => {
+        it('should get all file and filter them', async () => {
```
Comment (MichaelSimkin) [inline tests/incoming.test.ts:75]:
```suggestion
        it('should get all files by filters', async () => {
```

File: tsconfig.json [hunk 18]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "lib": ["ES2021"],
```
Comment (MichaelSimkin) [inline tsconfig.json:3]: What is this?

File: src/express/incoming/manager.ts (hunk 1 above)
Comment (sternomer) [reply src/express/incoming/manager.ts:9]: ok

**Review by yarinBenisty — APPROVED** [review]
nice!

**Pull request merged**